src/test/java/pianokeys/CompositionViewTest.java

```diff
@@ -15,7 +16,10 @@ public static void main(String[] args)
             frame.setSize(1600, 400); // Set size large enough for clear view
 
             Composition composition = new Composition();
-            final CompositionView compView = new CompositionView(composition);
+
+            Supplier<PianoController> controllerSupplier = () -> null;
```

Comment:
This is a bug. This should return a `mock()` instead of null. 
OR create a mock PianoController and return that in the Supplier.


```suggestion
            PianoController controller = mock();
            Supplier<PianoController> controllerSupplier = () -> controller;
```